### PR TITLE
Emergency fix for Vehicle Upgrade Modules!

### PR DESCRIPTION
**Pull request by jonahnm**
An emergency fix for Vehicle Upgrade Modules, fixes issues described in reddit posts: https://www.reddit.com/r/subnautica/comments/14o0x0x/anyone_know_what_this_is/?utm_source=share&utm_medium=ios_app&utm_name=ioscss&utm_content=2&utm_term=1
and
https://www.reddit.com/r/subnautica/comments/14ne2zz/seamoth_cant_move_after_savingreturning_to_game/?utm_source=share&utm_medium=ios_app&utm_name=ioscss&utm_content=2&utm_term=1
MERGE ASAP, THIS IS POSSIBLY AFFECTING ALL USERS!!!!!

### CODE REVIEW

**Comment by jonahnm** [comment]
I have read the CLA Document and I hereby sign the CLA.

**Comment by jonahnm** [comment]
recheck

**Comment by jonahnm** [comment]
Closes #425

**Comment by jonahnm** [comment]
closes #425

**Review by VELD-Dev — APPROVED** [review]
I was about to do that

File: Nautilus/Patchers/VehicleUpgradesPatcher.cs [hunk 1]
```diff
@@ -239,7 +239,7 @@ private static void OnUpgradeModuleExoHull(Exosuit __instance, int slotID, TechT
             if (Exosuit.crushDepths.TryGetValue(techTypeInSlot, out depthToCheck) && depthToCheck > newCrushDepth)
             {
                 newCrushDepth = depthToCheck;
-                if (ExosuitUpgradeModules[techTypeInSlot].TryGetGadget(out UpgradeModuleGadget mdlGadget))
+                if (ExosuitUpgradeModules.TryGetValue(techTypeInSlot, out _) && ExosuitUpgradeModules[techTypeInSlot].TryGetGadget(out UpgradeModuleGadget mdlGadget))
```
Comment (EldritchCarMaker) [inline Nautilus/Patchers/VehicleUpgradesPatcher.cs]: Why use a discard variable and not just use the output from the try get value?
 ``` 
if (ExosuitUpgradeModules.TryGetValue(techTypeInSlot, out upgrade) && upgrade.TryGetGadget(out UpgradeModuleGadget mdlGadget))
```
 this way you aren't unnecessarily getting the value twice

Comment (jonahnm) [reply Nautilus/Patchers/VehicleUpgradesPatcher.cs]: good question, lemme change that.

Comment (jonahnm) [reply Nautilus/Patchers/VehicleUpgradesPatcher.cs]: Did this.

**Review by VELD-Dev — APPROVED** [review]
that's better now yeah

**Review by 32Kallies — APPROVED** [review]

**Pull request merged**